miden-lib/asm/tx_prologue.masm

```diff
@@ -0,0 +1,255 @@
+# MEMORY OFFSETS
+# =================================================================================================
+const.GLOBAL_INPUTS=10
+const.ACCOUNT_DATA=100
+const.CONSUMED_NOTE_DATA=1000
+
+# PUBLIC INPUTS
+# =================================================================================================
+
+#! process_public_inputs[
+#!    block_hash: Digest,
+#!    account_id: felt,
+#!    initial_account_hash: Digest,
+#!    nullifier_commitment: Digest
+#!    ] -> []
+#!
+#! The process public inputs procedure is executed at the beginning of the transaction prologue and is responsible for
+#! storing public stack inputs in the global inputs section of memory.
+proc.process_public_inputs
+    # store block hash in global inputs
+    push.GLOBAL_INPUTS mem_storew dropw
+
+    # store AccountId in global inputs
+    push.GLOBAL_INPUTS.1 add mem_store
+
+    # store initial account hash in global inputs
+    push.GLOBAL_INPUTS.2 add mem_storew dropw
+
+    # store nullifier commitment in memory.  This is computed as a sequential hash of all (nullifier, script_root) pairs.
+    push.GLOBAL_INPUTS.3 add mem_storew dropw
+end
+
+# ACCOUNT DATA
+# =================================================================================================
+
+#! process_account_data() -> []
+#!
+#! This procedure is responsible for reading and un-hashing account data from the advice provider and subsequently
+#! computing the account hash.
+proc.process_account_data
+    # compute the address to store the account data
+    push.ACCOUNT_DATA.1 add
+
+    # prepare the stack for absorbing account data
+    padw padw padw
+
+    # absorb the account data
+    adv_pipe adv_pipe
+
+    # extract digest from hasher rate elements (h_0, ..., h_3)
+    dropw swapw dropw
+
+    # store the account hash in memory
+    push.ACCOUNT_DATA mem_storew
```

Comment:
Don't we already have account hash in memory (at `GLOBAL_INPUTS + 1`)? Maybe we can just define `ACCT_HASH_PTR` and store account hash there just once?